numpy/core/src/multiarray/ctors.c

```diff
@@ -1806,7 +1812,29 @@ PyArray_GetArrayParamsFromObject(PyObject *op,
             return 0;
         }
         /* If object arrays are forced */
-        if (is_object) {
+        if (is_object != DISCOVERED_OK) {
+            static PyObject * visibleDeprecationWarning = NULL;
+            if (visibleDeprecationWarning == NULL) {
+                npy_cache_import(
+                    "numpy", "VisibleDeprecationWarning",
+                    &visibleDeprecationWarning);
+                if (visibleDeprecationWarning == NULL) {
+                    return -1;
+                }
+            }
+            if (is_object == DISCOVERED_RAGGED && requested_dtype == NULL) {
+                /* NumPy 1.19, 2019-11-01 */
+                if (PyErr_WarnEx(visibleDeprecationWarning, "Creating an "
+                    " ndarray from ragged nested sequences (which is a "
```

Comment:
```suggestion
                    "ndarray from ragged nested sequences (which is a "
```

Reply:
Ack, forgot to press submit: Maybe rather than the bracketed explanation, put in the example: `such as '[[1], [2, 3]]' where '[2, 3]' is longer than '[1]'`? I would hope that is more useful.

Not sure though, and I am fine with it as is (we can always revise as well)

Reply:
fixing